lib/generators/matestack/component/component_generator.rb

```diff
@@ -0,0 +1,27 @@
+module Matestack
+  module Generators
+    class ComponentGenerator < Rails::Generators::NamedBase
+      source_root File.expand_path('templates', __dir__)
+
+      class_option :dynamic, type: :boolean, default: false
+      class_option :scss, type: :boolean, default: false
+      class_option :haml, type: :boolean, default: false
+      class_option :namespace, type: :string
+
+      def namespace
+        options[:namespace]
+      end
+
+      def create_component
+        @dynamic = options[:dynamic]
+
+        # Future: Check for matestack-compatible namespacing!
+
+        template 'app/matestack/components/%namespace%/%file_name%.rb.tt'
+        template 'app/matestack/components/%namespace%/%file_name%.js.tt' if options[:dynamic]
+        template 'app/matestack/components/%namespace%/%file_name%.scss.tt' if options[:scss]
+        template 'app/matestack/components/%namespace%/%file_name%.haml.tt' if options[:haml]
```

Comment:
I'm wondering if this works without a namespace or if it creates a path containing two `/`s? Probably covered by the tests, though ;) if not, add a `/` to `options[:namespace]` in line 12?

Reply:
It does work without a namespace ;) The `template` method creates the file directly in the `components` folder if no namespace is given. I suppose multiple slashes get reduced to a single one.

Reply:
I guess that's what they call magic :D nice!